File: old/BottomPop-up-box-master/maskDemo/ViewController.m
//
//  ViewController.m
//  maskDemo
//
//  Created by pengpeng yan on 2016/11/3.
//  Copyright © 2016年 pengpeng yan. All rights reserved.
//
#import "CustomerView.h"
#import "ViewController.h"

@interface ViewController ()
/** 自定义 */
@property (nonatomic ,strong) UIButton *btn;
/** 自定义view */
@property (nonatomic ,strong) CustomerView *custView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    [self loutSubViews];
    
}

- (void)btnAction:(UIButton *)btn{
    self.custView = [[CustomerView alloc] initWithHeight:200];
    [self.custView showViewController:self];
}

- (UIButton *)btn{
    if (!_btn) {
        _btn = [[UIButton alloc] init];
        _btn.frame = CGRectMake(0, 64, 50, 20);
        [_btn setTitle:@"an" forState:UIControlStateNormal];
        [_btn setBackgroundColor:[UIColor grayColor]];
        [_btn addTarget:self action:@selector(btnAction:) forControlEvents:UIControlEventTouchUpInside];
        
          }
    return _btn;
}

- (void)loutSubViews{
    [self.view addSubview:self.btn];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Demo-master/ButtonBreakDemo/ButtonBreakDemo/ViewController.swift
//
//  ViewController.swift
//  ButtonBreakDemo
//
//  Created by Jason on 16/6/22.
//  Copyright © 2016年 littleBoy. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        
        let btn = UIButton(type: UIButtonType.Custom)
        btn.frame = CGRectMake(0, 100, 150, 30)
        btn.backgroundColor = UIColor.redColor()
        view.addSubview(btn)
        btn.breakTime = 2.0
        btn.addTarget(self, action: "btnClick1", forControlEvents: UIControlEvents.TouchUpInside)

    
        let btn2 = UIButtonBreak(type: UIButtonType.Custom)
        btn2.time = 2;
        btn2.frame = CGRectMake(0, 150, 150, 30)
        btn2.backgroundColor = UIColor.redColor()
        view.addSubview(btn2)
        btn2.addTarget(self, action: "btnClick2", forControlEvents: UIControlEvents.TouchUpInside)
        
        
    }
    
    override func viewWillAppear(animated: Bool) {
        
    }
    
    func btnClick1() {
        print("\(NSDate())")
    }
    
    func btnClick2() {
        print("\(NSDate())")
    }

    

}

